### PR TITLE
Polyhedron Demo: Popup for OFF

### PR DESCRIPTION
**Pull request by maxGimeno**
## Summary of Changes

Add a pop-up to ask if the user wants to export the internal properties in the off plugin.

## Release Management

* Affected package(s):Polyhedron_demo

### CODE REVIEW

**Comment by lrineau** [comment]
What are "internal properties"?

**Comment by maxGimeno** [comment]
It should be normals and colors, but I just realized that colors don't work.

**Comment by lrineau** [comment]
For the UX, that popup is not good at all: on most cases the surface mesh will not have any internal property anyway, and even if there are, the popup should name them, so that the user can understand and choose to save them or not.

As a user, I would freeze in front of that popup: what if I click on "yes"? what will happen. And what if I click on "no"? is there any chance that I lose something?

**Comment by maxGimeno** [comment]
>  on most cases the surface mesh will not have any internal property anyway

The surface mesh will ALWAYS have internal properties for its normals, that is actually the real point of this PR.

> As a user, I would freeze in front of that popup: what if I click on "yes"? what will happen. And what if I click on "no"? is there any chance that I lose something?

I'll see what I can do about listing the concerned properties in the popup then.

**Comment by maxGimeno** [comment]
New pop-up looks like this :
image

File: Polyhedron/demo/Polyhedron/Scene_surface_mesh_item.cpp [hunk 1]
```diff
@@ -1513,11 +1513,42 @@ bool Scene_surface_mesh_item::hasPatchIds()
...
+    }
+  QString message = tr("Do you wish to export the following internal properties ? \n");
+  for(auto s : internal_properties)
+  {
+    message.append(tr(" -"));
+    message.append(s.c_str());
+    message.append(tr("\n"));
+  }
+  QMessageBox::StandardButton save_internal_properties =
+      QMessageBox::question(CGAL::Three::Three::mainWindow(), tr("Internal Properties"), message);
```
Comment (sloriot) [inline Polyhedron/demo/Polyhedron/Scene_surface_mesh_item.cpp:1539]:
```suggestion
      QMessageBox::question(CGAL::Three::Three::mainWindow(), tr("Save Properties?"), message);
```

File: Polyhedron/demo/Polyhedron/Scene_surface_mesh_item.cpp [hunk 2]
```diff
@@ -1513,11 +1513,42 @@ bool Scene_surface_mesh_item::hasPatchIds()
...
 {
+  std::vector<std::string> internal_properties;
+  std::vector<std::string> vprop = d->smesh_->properties<vertex_descriptor>();
+  std::vector<std::string> fprop = d->smesh_->properties<face_descriptor>();
+  for(auto s : vprop)
+     if (s.compare("v:normal") == 0
+         || s.compare("v:texcoord") == 0
+         || s.compare("v:color") == 0)
+    {
+       internal_properties.push_back(s);
```
Comment (sloriot) [inline Polyhedron/demo/Polyhedron/Scene_surface_mesh_item.cpp:1524]: You can fill message directly here and use a user friendly text like "Vertex Normals"...

File: Polyhedron/demo/Polyhedron/Scene_surface_mesh_item.cpp [hunk 3]
```diff
@@ -1513,11 +1513,42 @@ bool Scene_surface_mesh_item::hasPatchIds()
...
+    }
+  for(auto s : fprop)
+    if(s.compare("f:color") == 0)
+    {
+      internal_properties.push_back(s);
+    }
+  QString message = tr("Do you wish to export the following internal properties ? \n");
```
Comment (sloriot) [inline Polyhedron/demo/Polyhedron/Scene_surface_mesh_item.cpp:1531]:
```suggestion
  QString message = tr("Do you want to save the following properties? \n");
```

**Comment by maxGimeno** [comment]
image

**Comment by sloriot** [comment]
Either "Colors" or "Normal"  to be consistent :)

**Review by sloriot — APPROVED** [review]

**Comment by sloriot** [comment]
Successfully tested in [5.4-Ic-8](https://cgal.geometryfactory.com/CGAL/testsuite/results-5.4-Ic-8.shtml)

**Pull request merged**